lib/vscode_helper.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright 2021 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'dart:convert';
+
+import 'package:file/file.dart';
+import 'package:flutter_tools/src/convert.dart';
+import 'package:flutter_tools/src/globals.dart' as globals;
+import 'package:flutter_tools/src/project.dart';
+
+const String kConfigNameAttach = 'Attach (flutter-tizen)';
+
+final RegExp _commentFormat =
+    RegExp(r'(:?)(/\*([^*]|[\r\n]|(\*+([^*/]|[\r\n])))*\*+/|//.*\n?)');
+
+String _removeComments(String jsonString) {
+  return jsonString.replaceAllMapped(
+    _commentFormat,
+    (Match match) {
+      // "aaa://bbb" contains "//" but is not a comment.
+      return match.group(1) == ':' ? match.group(0) : '';
+    },
+  ).trim();
+}
+
+void updateLaunchJsonFile(FlutterProject project, Uri observatoryUri) {
+  Directory vscodeDir;
+  String cwd = r'${workspaceFolder}';
+  if (project.directory.basename == 'example') {
+    final FlutterProject parentProject =
+        FlutterProject.fromDirectory(project.directory.parent);
+    if (parentProject.isPlugin || parentProject.isModule) {
+      vscodeDir = parentProject.directory.childDirectory('.vscode')
+        ..createSync(recursive: true);
+      cwd += '/example';
+    }
+  }
+  vscodeDir ??= project.directory.childDirectory('.vscode')
+    ..createSync(recursive: true);
+
+  final File launchJsonFile = vscodeDir.childFile('launch.json');
+  String jsonString = '';
+  if (launchJsonFile.existsSync()) {
+    // Comments are allowed in launch.json file. Remove them to prevent
+    // decoding errors.
+    jsonString = _removeComments(launchJsonFile.readAsStringSync());
```

Comment:
just question :)  Is there any problem to parse comments with  `jsonDecoder` of `dart:convert` ?

Reply:
Comments are not allowed in standard JSON files.

![image](https://user-images.githubusercontent.com/40190588/129307238-7b383552-03cb-488f-a3cd-86013e1ae5db.png)

`jsonDecode` will throw an error when a comment is encountered.

Reply:
Oh.. I see. that's special thing about launch.json....  😲

Reply:
Maybe I could create my own SAX-like JSON parser to keep the original comments, but it's not very easy :(